test/functional/test_framework/test_node.py

```diff
@@ -197,6 +197,19 @@ def wait_for_rpc_connection(self):
             time.sleep(1.0 / poll_per_s)
         self._raise_assertion_error("Unable to connect to bitcoind")
 
+    def generate(self, nblocks, maxtries=1000000):
+        self.log.debug("TestNode.generate() dispatches `generate` call to `generatetoaddress`")
+        # Try to import the node's deterministic private key. This is a no-op if the private key
+        # has already been imported.
+        try:
+            self.rpc.importprivkey(privkey=self.get_deterministic_priv_key().key, label='coinbase', rescan=False)
+        except JSONRPCException as e:
+            assert str(e).startswith('Method not found') or \
```

Comment:
I'm not sure I understand this assertion, do we want here  to blanket-ignore exceptions matching these strings?

Reply:
Only accept `importprivkey` failure if:
 - wallet is disabled
 - there are multiple wallets to import to
 - wallet is locked

In either case the outcome is the impossibility of spending the coinbase output. And if that's relevant to the test then it would fail for not having enough coins to spend.

Reply:
@promag is right. This call to `importprivkey` can fail for the three reasons listed.

I think that this is a bit too much magic behind the scenes though. I've tidied up the deterministic priv key import behaviour in another branch here: https://github.com/jnewbery/bitcoin/tree/deprecate_generate2. Rather than complicate this PR and add review burden here, are you happy to leave this as is for now and take the final commit in https://github.com/jnewbery/bitcoin/tree/deprecate_generate2 as a follow-up PR?

Reply:
@jnewbery yes, I'm okay with that!
though for this PR, please add a comment w/ the contents of @promag's post or something like that, so that people reading the code and wondering the same understand

Reply:
Comment added.